### DISCUSSION TITLE
Possible to use built in column sorting without internal editing and using ReadData?

### DISCUSSION THREAD

**Discussion by Andrew-Seagle** [Megabit/Blazorise#3507] [Q&A] [answered]
I was wondering if there was a way to have the built in sorting in a data grid still work if I've disabled internal editing and while using read data. I may have incorrectly assumed that "editing" would mean only the CRUD functions, but it seems sort is included in that, or am I missing something?

**Comment by stsrki** [comment, marked as answer]
Sorting should work even when editing is disabled. If it's not then it means it is a bug.

**Reply by Andrew-Seagle** [reply]
After messing around with it a bit more, it seems to be the use of ReadData that is stopping the sorting.
Is that the expected behavior, or am I misunderstanding the data grid's interaction with the collection that it's given?

**Reply by David-Moreira** [reply]
Hello,
With `ReadData` active, your in control of the data loading, paging, filtering and sorting, there should be information in the arguments, so you can do these activities. This means, that the data should arrive already according to current pages / number of items, based on filter and sorting that the `DataGrid` requested.

We plan on making some kind of API available that abstracts these features for you. 
Tracked here: https://github.com/Megabit/Blazorise/issues/3445

There's an example of a simple implementation on our demo:
https://bootstrapdemo.blazorise.com/tests/datagrid
You have to select `"Large Data"`

Code: 
https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/DataGrid/DataGridPage.razor.cs
Line 154 for our `ReadData` implementation.

**Reply by Andrew-Seagle** [reply]
That makes sense. An API would be an amazing feature for this, I hope the implementation of that goes well for you guys.
Thank you for pointing me to an such a good example!
